tcf_website/views/review.py

```diff
@@ -47,49 +47,52 @@ def new_review(request):
         form = ReviewForm(request.POST)
         if form.is_valid():
             try:
-                course_code = request.POST['course']
-                mnemonic, number = course_code.split()
-                course = Course.objects.get(
-                    subdepartment__mnemonic=mnemonic, number=int(number))
+                course_id = request.POST['courseID']
+                course = Course.objects.get(id=int(course_id))
 
-                instructor_name = request.POST['instructor']
-                first, last = instructor_name.split()
-                instructor = Instructor.objects.get(
-                    first_name=first, last_name=last)
+                instructor_id = request.POST['instructor']
+                instructor = Instructor.objects.get(id=int(instructor_id))
 
-                semester_str = request.POST['semester']
-                season, year = semester_str.split()
-                semester = Semester.objects.get(
-                    season=season.upper(), year=int(year))
+                semester_id = request.POST['semester']
+                semester = Semester.objects.get(id=int(semester_id))
+
+                hours_reading = int(request.POST['hoursReading'])
+                hours_writing = int(request.POST['hoursWriting'])
+                hours_group = int(request.POST['hoursGroupwork'])
+                hours_other = int(request.POST['hoursOther'])
+                total_hours = hours_reading + hours_writing + hours_group + hours_other
 
                 Review.objects.create(
                     user=request.user,
                     course=course,
                     semester=semester,
                     instructor=instructor,
                     text=request.POST['reviewText'],
-                    instructor_rating=int(
-                        request.POST['instructorRating']),
+                    instructor_rating=int(request.POST['instructorRating']),
+                    enjoyability=int(request.POST['enjoyability']),
                     difficulty=int(request.POST['difficulty']),
-                    recommendability=int(
-                        request.POST['recommendability']),
-                    hours_per_week=int(request.POST['hours']),
+                    recommendability=int(request.POST['recommendability']),
+                    amount_reading=hours_reading,
+                    amount_writing=hours_writing,
+                    amount_group=hours_group,
+                    amount_homework=hours_other,
+                    hours_per_week=total_hours
                 )
 
                 messages.add_message(
                     request,
                     messages.SUCCESS,
                     'Successfully reviewed ' +
-                    str(course_code) +
-                    '!')
+                    str(course) + '!')
                 return redirect('reviews')
             except KeyError as err:
```

Comment:
What if another error (besides `KeyError`) occurs? Off the top of my head, `Course.DoesNotExist`, `Instructor.DoesNotExist`, and `TypeError` (when casting to int) can occur here.

Reply:
Okay, I can change this to catch all errors for now. @alex-shen1 was planning to add more robust backend validation later.